### PR TITLE
mgr/dashboard: run cephadm-backend e2e tests with KCLI

### PR DESCRIPTION
**Pull request by alfonsomthd**
Script to run E2E tests with cephadm as orchestrator backend.
To be run in this jenkins job: https://github.com/ceph/ceph-build/pull/1842

Fixes: https://tracker.ceph.com/issues/51300

Signed-off-by: Alfonso Martínez <almartin@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Review by jmolmo — APPROVED** [review]
This is awesome!!. Thanks Alfonso!!

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

File: src/pybind/mgr/dashboard/ci/cephadm/bootstrap-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+mon_ip=$(ifconfig eth0  | grep 'inet ' | awk '{ print $2}')
+{% if ceph_dev_folder is defined %}
+  cephadm bootstrap --mon-ip $mon_ip --initial-dashboard-password {{ admin_password }} --allow-fqdn-hostname --dashboard-password-noupdate --shared_ceph_folder /mnt/{{ ceph_dev_folder }}
+{% else %}
+  cephadm bootstrap --mon-ip $mon_ip --initial-dashboard-password {{ admin_password }} --allow-fqdn-hostname --dashboard-password-noupdate
+{% endif %}
+fsid=$(cat /etc/ceph/ceph.conf | grep fsid | awk '{ print $3}')
+{% for number in range(1, nodes) %}
+  ssh-copy-id -f -i /etc/ceph/ceph.pub  -o StrictHostKeyChecking=no root@{{ prefix }}-node-0{{ number }}.{{ domain }}
+  #cephadm shell --fsid $fsid -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring ceph orch host add {{ prefix }}-node-0{{ number }}.{{ domain }}
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/bootstrap-cluster.sh:22]: ?

File: src/pybind/mgr/dashboard/ci/cephadm/ceph_cluster.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+parameters:
+ nodes: 3
+ pool: default
+ network: default
+ domain: cephlab.com
+ prefix: ceph
+ numcpus: 1
+ memory: 2048
+ image: fedora34
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/ceph_cluster.yml:9]: ?
```suggestion
 image: centos8
```

**Review by epuertat — APPROVED** [review]
LGMT! Good job @alfonsomthd !

**Review by nizamial09 — APPROVED** [review]
LGTM! Tested locally and verified

File: src/pybind/mgr/dashboard/ci/cephadm/ceph_cluster.yml (hunk 2 above)
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/ci/cephadm/ceph_cluster.yml:9]: @epuertat Centos8 does not support 9p filesystem type which is required for sharing the host folder.

File: doc/dev/developer_guide/dash-devel.rst [hunk 3]
```diff
@@ -423,6 +423,26 @@ Note:
   When using docker, as your device, you might need to run the script with sudo
   permissions.
 
+run-cephadm-e2e-tests.sh
+.........................
+
+``run-cephadm-e2e-tests.sh`` runs a subset of E2E tests to verify that the Dashboard and cephadm as
+Orchestrator backend behave correctly.
+
+Prerequisites: you need to install `KCLI
```
[the remark this answers is not in the document]
Comment (alfonsomthd) [reply doc/dev/developer_guide/dash-devel.rst:432]: @nizamial09 the jenkins job is currently using @jmolmo 's image.
Currently @jmolmo is working on solving some issues and when appropriate we'll switch to the appropriate KCLi version.

[the remark this answers is not in the document]
Comment (nizamial09) [reply doc/dev/developer_guide/dash-devel.rst:432]: @alfonsomthd sounds good to me. Thank you

File: src/pybind/mgr/dashboard/ci/cephadm/bootstrap-cluster.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{% if ceph_dev_folder is defined %}
+  cephadm bootstrap --mon-ip $mon_ip --initial-dashboard-password {{ admin_password }} --allow-fqdn-hostname --dashboard-password-noupdate --shared_ceph_folder /mnt/{{ ceph_dev_folder }}
+{% else %}
+  cephadm bootstrap --mon-ip $mon_ip --initial-dashboard-password {{ admin_password }} --allow-fqdn-hostname --dashboard-password-noupdate
+{% endif %}
+fsid=$(cat /etc/ceph/ceph.conf | grep fsid | awk '{ print $3}')
+{% for number in range(1, nodes) %}
+  ssh-copy-id -f -i /etc/ceph/ceph.pub  -o StrictHostKeyChecking=no root@{{ prefix }}-node-0{{ number }}.{{ domain }}
+{% endfor %}
+cephadm shell --fsid $fsid -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring ceph orch apply osd --all-available-devices
```
[the remark this answers is not in the document]
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/ci/cephadm/bootstrap-cluster.sh:23]: Done.

**Pull request merged**